components/scream/src/share/io/scream_scorpio_interface.cpp

```diff
@@ -121,8 +121,8 @@ void grid_read_data_array(const std::string &filename, const std::string &varnam
   grid_read_data_array_c2f(filename.c_str(),varname.c_str(),time_index,hbuf);
 }
 /* ----------------------------------------------------------------- */
-void grid_write_data_array(const std::string &filename, const std::string &varname, const Real* hbuf) {
-  grid_write_data_array_c2f_real(filename.c_str(),varname.c_str(),hbuf);
+void grid_write_data_array(const std::string &filename, const std::string &varname, void *hbuf) {
```

Comment:
@bartgol , I noticed in the scorpio Slack space you were asking about something that may be related to this change.  So I thought I would ping you just in case you are working on something similar.

Reply:
Last weekend, I was playing around with scorpio C (to bypass flib). The C interface uses a non-const buffer, which forces some const cast. In our master, however, we fwd the call to F90, where const doesn't matter, so we don't need this change.

Reply:
I think we still need this change here.  I added it to allow us to write both REAL and INT output.  Currently in master we can only write REAL.  I copied the syntax from read_array which can support both formats.

Reply:
Ah, sorry, yes, the `void` part you do need it. But you should keep `const` in front, to allow to be able to write read-only fields.